clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -134,6 +134,10 @@ static Atom *analyzeAllocExpr(Expr *E, Constraints &CS, QualType &ArgTy) {
 
   // Look for sizeof(X); return Arr or Ptr if found
   for (Expr *Ex: Exprs) {
+    //normalize the expression if part of an implicit cast
+    if(ImplicitCastExpr *imp = dyn_cast<ImplicitCastExpr>(Ex)) {
+      Ex = imp->getSubExprAsWritten();
+    }
```

Comment:
I would change these four lines to `Ex = Ex->IgnoreParenImpCasts();` That will get rid of parentheses and implicit casts.

Reply:
This was actually the first thing I tried out, but abandoned it because after doing this conversion, I used `getStmtClassName`, and it still returned ImplictCast, so I assumed it hadn't done anything. Good to know it can be replaced though. Did you happen to look at the conversation about making sure `z` doesn't get converted when running without `alltypes`? The only way I can get this to happen is with the weird alltypes hacky fix.